lib/metasploit/framework/aws/client.rb

```diff
@@ -0,0 +1,179 @@
+require 'openssl'
+
+module Metasploit
+  module Framework
+    module Aws
+      module Client
```

Comment:
It would be excellent if there were specs for this client, or at least the critical parts.  Some seem easily testable, others not.  See what you can do and we can lend a hand.